### ISSUE TITLE
Streaming with demonstration server stops, node process at 100% CPU

### ISSUE THREAD

**Issue by Pheo-Player** [fabslab/aurora-websocket#2]
I tried out the example client code + node server (with only the `audioFolder` and the `port` changed).
Streaming always stops at random points between a few seconds after playback starts, and about half a minute. I started logging when data events are emitted, and noticed it stops quite quickly, before sending the file. Looking at the output of `top`, I saw my suspicions confirmed: the node process running the server shoots to ~100% CPU usage in less than a second, stops sending anything, and audio abruptly ends as soon as the received data has been played back on the client.

System info:
Mac OS X 10.9.5
node `v0.10.30`

I tried this with different files, and it seems this is actually the same issue as https://github.com/fsbdev/aurora-websocket/issues/1. With a FLAC file, I got no further errors on the client, but with an MP3, I also get an `UnderflowError` _as soon as the server hits 100%_ and stops sending anything.

P.S. Thank you for the great work so far, without this repo existing I might have lost the vision for my project :smile:

**Comment by Pheo-Player** [comment]
This might actually be an issue with the `throttle` library.

I noticed that when I removed stream throttling, the node process did not mess up in this way (with the downside that this mostly ruins the point of the library, as is pointed out in the docs).

I then found this library: https://www.npmjs.org/package/stream-throttle and changed only two lines in the original server:

``` js
Throttle = require('stream-throttle').Throttle;
```

and

``` js
audioStream = fs.createReadStream(audioPath).pipe(new Throttle({ rate: 700 * 1024 }));
```

after doing this, everything works just perfect.

Keeping this issue open for now, as you might want to update your example server (or investigate further, but it seems that the problem is definitely not on the side of this library).
